Mathlib/FieldTheory/SeparableClosure.lean

```diff
@@ -0,0 +1,386 @@
+if it is separable over `F`. -/
+theorem le_separableClosure (L : IntermediateField F E) [IsSeparable F L] :
+    L ≤ separableClosure F E := le_separableClosure' F E (IsSeparable.separable F)
+
+/-- An intermediate field of `E / F` is contained in the (relative) separable closure of `E / F`
+if and only if it is separable over `F`. -/
+theorem le_separableClosure_iff (L : IntermediateField F E) :
+    L ≤ separableClosure F E ↔ IsSeparable F L :=
+  ⟨fun h ↦ (isSeparable_def _ _).2 fun x ↦ by simpa only [minpoly_eq] using h x.2,
+    fun _ ↦ le_separableClosure _ _ _⟩
+
+/-- The (relative) separable closure of the (relative) separable closure of `E / F` is equal to
+itself. -/
+theorem separableClosure.separableClosure_eq_bot :
+    separableClosure (separableClosure F E) E = ⊥ := bot_unique fun x hx ↦ by
+  rw [mem_separableClosure_iff, ← isSeparable_adjoin_simple_iff_separable] at hx
+  set L := separableClosure F E
+  haveI : IsSeparable F (L⟮x⟯.restrictScalars F) := IsSeparable.trans F L L⟮x⟯
+  exact mem_bot.2 ⟨⟨x, (mem_separableClosure_iff F E).2 (separable_of_mem_isSeparable F E <|
+    show x ∈ L⟮x⟯.restrictScalars F from mem_adjoin_simple_self L x)⟩, rfl⟩
+
+/-- The normal closure of the (relative) separable closure of `E / F` is equal to itself. -/
+theorem separableClosure.normalClosure_eq_self :
+    normalClosure F (separableClosure F E) E = separableClosure F E := by
+  apply le_antisymm (normalClosure_le_iff.2 fun i ↦ ?_) (le_normalClosure _)
+  let i' : (separableClosure F E) ≃ₐ[F] i.fieldRange := AlgEquiv.ofInjectiveField i
+  haveI := i'.isSeparable
+  exact le_separableClosure F E _
+
+/-- If `E` is normal over `F`, then the (relative) separable closure of `E / F` is also normal
+over `F`. -/
+instance separableClosure.normal [Normal F E] : Normal F (separableClosure F E) := by
+  rw [← separableClosure.normalClosure_eq_self]
+  exact normalClosure.normal F _ E
+
+/-- If `E` is separably closed, then the (relative) separable closure of `E / F` is a
+separable closure of `F`. -/
+instance separableClosure.isSepClosure [IsSepClosed E] : IsSepClosure F (separableClosure F E) := by
+  refine ⟨IsSepClosed.of_exists_root _ fun p _ hirr hsep ↦ ?_, isSeparable F E⟩
+  obtain ⟨x, hx⟩ := IsSepClosed.exists_aeval_eq_zero E p (degree_pos_of_irreducible hirr).ne' hsep
+  haveI := (isSeparable_adjoin_simple_iff_separable _ E).2 <| hsep.of_dvd <| minpoly.dvd _ x hx
+  let L := separableClosure F E
+  haveI : IsSeparable F (restrictScalars F L⟮x⟯) := IsSeparable.trans F L L⟮x⟯
+  have : x ∈ restrictScalars F L⟮x⟯ := mem_adjoin_simple_self _ x
+  use ⟨x, le_separableClosure F E (restrictScalars F L⟮x⟯) this⟩
+  apply_fun algebraMap L E using (algebraMap L E).injective
+  rwa [map_zero, ← aeval_algebraMap_apply_eq_algebraMap_eval]
+
+/-- The (absolute) separable closure is defined to be the (relative) separable closure inside the
+algebraic closure. -/
+abbrev SeparableClosure : Type _ := separableClosure F (AlgebraicClosure F)
```

Comment:
Can you add some API around this or at least add a todo?
For example (restating) that `IsGalois F (SeparableClosure F)`
Or the fact that every separable extension embeds into it.

Reply:
> For example (restating) that `IsGalois F (SeparableClosure F)`

This can by synth automatically from the coming `separableClosure.isGalois`.

> Or the fact that every separable extension embeds into it.

This is [IsSepClosed.lift](https://leanprover-community.github.io/mathlib4_docs/Mathlib/FieldTheory/IsSepClosed.html#IsSepClosed.lift).

Reply:
I see. Is the instance `IsSepClosure (SeparableClosure F)` coming in this PR?

Reply:
I found it. Thanks for the updated docstring.